fastlane/actions/get_latest_github_release.rb

```diff
@@ -0,0 +1,170 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      GET_LATEST_GITHUB_RELEASE_CUSTOM_VALUE = :GET_LATEST_GITHUB_RELEASE_CUSTOM_VALUE
+    end
+
+    class GetLatestGithubReleaseAction < Action
+      def self.run(params)
+        UI.message("Getting latest release on GitHub (#{params[:server_url]}/#{params[:url]})")
+
+        GithubApiAction.run(
+          server_url: params[:server_url],
+          api_token: params[:api_token],
+          api_bearer: params[:api_bearer],
+          http_method: 'GET',
+          path: "repos/#{params[:url]}/releases",
+          error_handlers: {
+            404 => proc do |result|
+              UI.error("Repository #{params[:url]} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
+              return nil
+            end,
+            401 => proc do |result|
+              UI.error("You are not authorized to access #{params[:url]}, please make sure you provided a valid API token.")
+              return nil
+            end,
+            '*' => proc do |result|
+              UI.error("GitHub responded with #{result[:status]}:#{result[:body]}")
+              return nil
+            end
+          }
+        ) do |result|
+          json = result[:json]
+          json.each do |current|
+            Actions.lane_context[SharedValues::GET_GITHUB_RELEASE_INFO] = current
+            UI.message("Version #{current['tag_name']} is latest release live on GitHub.com 🚁")
+            return current
```

Comment:
Hmm I guess this code assumes the endpoint `/repos/repo-name/releases` returns releases in order... I guess this is true and I doubt it will change but not sure I like trusting on a 3rd party service like that... I checked the API docs and I don't see a sorting parameter, but it doesn't mention anything about the order either (https://docs.github.com/en/rest/releases/releases#list-releases)... Maybe we should sort the releases by date? 

That said, the number of releases increases over time, so it might get to a point where it's inefficient, so I'm ok if we want to trust github to not change this. We can probably notice when it's creating the changelog anyway. In that case, I would change the `each` on the `json` with a `first` assuming it's an `Array` which I'm not sure (https://apidock.com/ruby/Array/first)

Reply:
Hmmm that's true. I actually just realized too that sometimes the latest published release is actually not the latest release, like for example when publishing a hotfix on an older version. I think we should order by major, minor, patch instead. There's probably code written somewhere